src/Apps/Order2/Routes/Checkout/Hooks/useCheckoutTracking.ts

```diff
@@ -0,0 +1,143 @@
+import {
+  ActionType,
+  type ClickedCancelExpressCheckout,
+  type ClickedExpressCheckout,
+  type ErrorMessageViewed,
+  type ExpressCheckoutViewed,
+  type OwnerType,
+  type SubmittedOrder,
+} from "@artsy/cohesion"
+import { useAnalyticsContext } from "System/Hooks/useAnalyticsContext"
+import { useMemo } from "react"
+import { useTracking } from "react-tracking"
+
+export const useCheckoutTracking = () => {
+  const { trackEvent } = useTracking()
+  const analytics = useAnalyticsContext()
+  const contextPageOwnerId = analytics.contextPageOwnerId as string
+  const contextPageOwnerSlug = analytics.contextPageOwnerSlug as string
+  const contextPageOwnerType = analytics.contextPageOwnerType as OwnerType
+
+  const trackingCalls = useMemo(() => {
+    return {
+      submittedOrder: ({
+        order,
+        walletType,
+      }: {
+        order: { source: string }
```

Comment:
Better here would be to just pass `source` (and ditto elsewhere); no need to nest it within `order` as the callsite will provide the necessary context -- eg, 

```tsx
tracking.submittedOrder({
  source: order.source,
  foo: order.foo
  bar: order.bar
})
```